lib/elixir/src/elixir_expand.erl

```diff
@@ -223,12 +223,6 @@ expand({fn, Meta, Pairs}, E) ->
 
 %% Case/Receive/Try
 
-expand({'cond', Meta, [Opts]}, E) ->
-  assert_no_match_or_guard_scope(Meta, "cond", E),
```

Comment:
We should assert this in the Elixir code as well.

Reply:
Done in a21757d. Although, the error is now an ArgumentError not a CompileError